src/Application/Console/Formatters/FormatResolver.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Application\Console\Formatters;
+
+use InvalidArgumentException;
+use NunoMaduro\PhpInsights\Application\Console\Contracts\Formatter;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @internal
+ */
+final class FormatResolver
+{
+    /**
+     * @var array<string>
+     */
+    private static $formatters = [
+        'console' => Console::class,
+        'json' => Json::class,
+    ];
+
+    public static function resolve(
+        InputInterface $input,
+        OutputInterface $output
+    ): Formatter
+    {
+        $requestedFormat = $input->getOption('format');
+
+        if (! is_string($requestedFormat)) {
+            throw new InvalidArgumentException(
+                "Format has to be a string."
+            );
+        }
+
+        $requestedFormat = strtolower($requestedFormat);
+
+        $formatter = self::$formatters[$requestedFormat] ?? Console::class;
```

Comment:
:thinking: Shall we check if the format option entered is valid  (with array_key_exist) and throw exception instead ?

Reply:
Yeah was a little unsure about this myself.
Maybe show a message about invalid format, using fallback instead?

Reply:
Yeah, agree with that